### PR TITLE
Updated MonoGame to use the latest stable (2.6.3) SharpDX from NuGet

### PR DESCRIPTION
**Pull request by mattleibow**
Updated the code to support SharpDX version 2.6.3 on all Windows platforms, except UAP which is 3.0.
This PR must go in first otherwise the build will fail: https://github.com/Mono-Game/MonoGame.Dependencies/pull/59
These changes are minimal, either using existing WinRT code, or existing logic being moved into the ShaprDX types (SourceResolver)

### CODE REVIEW

File: MonoGame.Framework/Audio/SoundEffect.XAudio.cs [hunk 1]
```diff
@@ -73,15 +73,6 @@ internal static void InitializeSoundEffect()
...
                 {
-#if !WINRT && DEBUG
-                    try
-                    {
-                        //Fails if the XAudio2 SDK is not installed
-                        Device = new XAudio2(XAudio2Flags.DebugEngine, ProcessorSpecifier.DefaultProcessor);
-                        Device.StartEngine();
-                    }
-                    catch
-#endif
```
Comment (KonajuGames) [inline MonoGame.Framework/Audio/SoundEffect.XAudio.cs]: Is there no `XAudio2Flags.DebugEngine` any more?

**Comment by tomspilman** [comment]
@mattleibow - Mono-Game/MonoGame.Dependencies#59 is now merged.

File: MonoGame.Framework/Audio/SoundEffect.XAudio.cs (hunk 1 above)
Comment (tomspilman) [inline MonoGame.Framework/Audio/SoundEffect.XAudio.cs]: Yeah...  I would prefer to not loose this feature if possible.

Comment (mattleibow) [inline MonoGame.Framework/Audio/SoundEffect.XAudio.cs]: Seems to have been removed in Windows 8 desktop: http://sharpdx.org/documentation/api/t-sharpdx-xaudio2-xaudio2flags

**Comment by mattleibow** [comment]
@tomspilman should I update the dependencies submodule as well? It has changes to FreeImage.NET and OpenTK

**Comment by tomspilman** [comment]
> should I update the dependencies submodule as well? It has changes to FreeImage.NET and OpenTK

Humm...  are those changes merged yet @KonajuGames ?

**Comment by totallyeviljake** [comment]
I am able to build this PR using the current master of the dependencies project. I had a conflict in the dependencies project config on my last upstream merge. maybe team city had a similar problem and did not update the dependencies submodule.

**Comment by totallyeviljake** [comment]
For the Windows 8 build, my Jenkins reports:

                 [exec] C:\Windows\Microsoft.NET\Framework\v4.0.30319\Microsoft.Common.targets(1605,5): warning MSB3275: The primary reference "SharpDX.DXGI" could not be resolved because it has an indirect dependency on the assembly "SharpDX, Version=2.6.3.0, Culture=neutral, PublicKeyToken=b4dcf0f35e5521f1" which was built against the ".NETCore,Version=v4.5.1" framework. This is a higher version than the currently targeted framework ".NETCore,Version=v4.5". [c:\workspace\Cocos2D-XNA-Windows8\MonoGame\MonoGame.Framework\MonoGame.Framework.Windows8.csproj]

**Comment by KonajuGames** [comment]
> Humm... are those changes merged yet @KonajuGames ?

The submodule reference with FreeImage libs was updated as part of the texture content processing PR that has already been merged.

OpenTK was updated in Dependencies by @cra0zy earlier this week.  Does that need any MonoGame changes, @cra0zy ?

**Comment by cra0zy** [comment]
> Does that need any MonoGame changes, @cra0zy ?

No MonoGame changes needed for it, it just includes some bug fixes.

**Comment by tomspilman** [comment]
@mattleibow - Please go ahead and update the dependencies submodule now.

**Comment by mattleibow** [comment]
The build fails because of:

> The primary reference "SharpDX" could not be resolved because it was built against the ".NETCore,Version=v4.5.1" framework. 
> This is a higher version than the currently targeted framework ".NETCore,Version=v4.5".

We are going to have to use the later version of .NET. It is possible to target Windows 8 and .NET 4.5.1 at the same time. I think on line 277-278 (https://github.com/mono/MonoGame/blob/develop/Build/GenerateProject.CSharp.xslt#L277-L278) should become:
```xml
<xsl:when test="/Input/Generation/Platform = 'Windows8'">
    <TargetFrameworkVersion>v4.5.1</TargetFrameworkVersion>
</xsl:when>
```

**Comment by tomspilman** [comment]
I guess this means all end users will have to move to 4.5.1 too?  We would also need to update our template projects.

**Comment by nkast** [comment]
Any change to include this change to the WP8.0 libraries?
https://github.com/sharpdx/SharpDX/pull/541

Bug details here:
https://github.com/sharpdx/SharpDX/issues/451
https://github.com/mono/MonoGame/issues/2848

**Comment by Shqrdx** [comment]
4.5.1 ? why not 4.6 ?

**Comment by tomspilman** [comment]
This branch has conflicts and can't be merged.  @mattleibow - Can you merge in the latest changes?

**Comment by mattleibow** [comment]
@tomspilman done.

**Comment by tomspilman** [comment]
Ok merging this.

**Comment by tomspilman** [comment]
@mattleibow - FYI.  This PR caused issues with audio on Win7 platforms.  https://github.com/mono/MonoGame/issues/4045

**Pull request merged**